File: module_rdp/src/main/java/com/xiaoyv/rdp/screen/listener/RdpBroadcastReceiver.java
package com.xiaoyv.rdp.screen.listener;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.blankj.utilcode.util.LogUtils;
import com.freerdp.freerdpcore.application.GlobalApp;
import com.freerdp.freerdpcore.application.SessionState;

public class RdpBroadcastReceiver extends BroadcastReceiver {

    private  SessionState session;

    public RdpBroadcastReceiver(SessionState session) {
        this.session = session;
    }

    public void setSession(SessionState session) {
        this.session = session;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        // 仍然有一个有效的会话？
        if (session == null) {
            return;
        }

        // is this event for the current session?
        if (session.getInstance() != intent.getExtras().getLong(GlobalApp.EVENT_PARAM, -1))
            return;

        switch (intent.getExtras().getInt(GlobalApp.EVENT_TYPE, -1)) {
            case GlobalApp.FREERDP_EVENT_CONNECTION_SUCCESS:
                OnConnectionSuccess(context);
                break;

            case GlobalApp.FREERDP_EVENT_CONNECTION_FAILURE:
                OnConnectionFailure(context);
                break;
            case GlobalApp.FREERDP_EVENT_DISCONNECTED:
                OnDisconnected(context);
                break;
        }
    }

    private void OnConnectionSuccess(Context context) {
        LogUtils.e("OnConnectionSuccess");
    }

    private void OnConnectionFailure(Context context) {
        LogUtils.e("OnConnectionFailure");
    }

    private void OnDisconnected(Context context) {
        LogUtils.e("OnDisconnected");
    }
}

File: module_rdp/src/main/java/com/xiaoyv/rdp/screen/config/RdpBroadcastReceiver.kt
package com.xiaoyv.rdp.screen.config

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.freerdp.freerdpcore.application.RdpApp

/**
 * @author Admin
 */
class RdpBroadcastReceiver : BroadcastReceiver() {
    var onPrepareConnect: (context: Context) -> Unit = {}
    var onConnectionSuccess: (context: Context) -> Unit = {}
    var onConnectionFailure: (context: Context) -> Unit = {}
    var onDisconnecting: (context: Context) -> Unit = {}
    var onDisconnected: (context: Context) -> Unit = {}

    var currentInstance = -1L

    override fun onReceive(context: Context, intent: Intent?) {
        intent ?: return
        val extras = intent.extras ?: return
        val instance = extras.getLong(RdpApp.EVENT_PARAM, -1)
        val rdpEventType = extras.getInt(RdpApp.EVENT_TYPE, -1)

        // 判断是否为当前会话的事件
        if (currentInstance != instance) {
            return
        }

        // 判断事件类型
        when (rdpEventType) {
            RdpApp.FREERDP_EVENT_PREPARE_CONNECT -> onPrepareConnect.invoke(context)
            RdpApp.FREERDP_EVENT_CONNECTION_SUCCESS -> onConnectionSuccess.invoke(context)
            RdpApp.FREERDP_EVENT_CONNECTION_FAILURE -> onConnectionFailure.invoke(context)
            RdpApp.FREERDP_EVENT_DISCONNECTING -> onDisconnecting.invoke(context)
            RdpApp.FREERDP_EVENT_DISCONNECTED -> onDisconnected.invoke(context)
        }
    }
}